File: app/src/main/java/com/rockon999/android/leanbacklauncher/recommendations/NotificationsServiceV4.java
package com.rockon999.android.leanbacklauncher.recommendations;

import android.service.notification.StatusBarNotification;

import com.rockon999.android.leanbacklauncher.recommendations.GservicesRankerParameters.Factory;
import com.rockon999.android.tvrecommendations.service.BaseNotificationsService;

public class NotificationsServiceV4 extends BaseNotificationsService {
    private NotificationServiceDelegate mDelegate;

    public interface NotificationServiceDelegate {
        void onFetchExistingNotifications(StatusBarNotification[] statusBarNotificationArr);

        void onNotificationPosted(StatusBarNotification statusBarNotification);

        void onNotificationRemoved(StatusBarNotification statusBarNotification);
    }

    public NotificationsServiceV4() {
        super(false, new Factory());
    }

    public void onListenerConnected() {
        super.onListenerConnected();
    }

    protected void onFetchExistingNotifications(StatusBarNotification[] notifications) {
        if (isEnabled()) {
            super.onFetchExistingNotifications(notifications);
            if (this.mDelegate != null) {
                this.mDelegate.onFetchExistingNotifications(notifications);
            }
        }
    }

    public void onNotificationPosted(StatusBarNotification sbn) {
        if (isEnabled()) {
            super.onNotificationPosted(sbn);
            if (this.mDelegate != null) {
                this.mDelegate.onNotificationPosted(sbn);
            }
        }
    }

    public void onNotificationRemoved(StatusBarNotification sbn) {
        if (isEnabled()) {
            super.onNotificationRemoved(sbn);
            if (this.mDelegate != null) {
                this.mDelegate.onNotificationRemoved(sbn);
            }
        }
    }

    private boolean isEnabled() {
        return true;
    }
}

File: app/src/main/java/com/amazon/tv/leanbacklauncher/recommendations/NotificationsServiceV4.kt
package com.amazon.tv.leanbacklauncher.recommendations

import android.service.notification.StatusBarNotification
import android.util.Log
import com.amazon.tv.leanbacklauncher.BuildConfig
import com.amazon.tv.tvrecommendations.service.BaseNotificationsService

class NotificationsServiceV4 :
    BaseNotificationsService(false, GservicesRankerParameters.Factory()) {
    private val mDelegate: NotificationServiceDelegate? = null
    private val TAG =
        if (BuildConfig.DEBUG) ("*" + javaClass.simpleName).take(21) else javaClass.simpleName

    interface NotificationServiceDelegate {
        fun onFetchExistingNotifications(statusBarNotificationArr: Array<StatusBarNotification>?)
        fun onNotificationPosted(statusBarNotification: StatusBarNotification?)
        fun onNotificationRemoved(statusBarNotification: StatusBarNotification?)
    }

    override fun onListenerConnected() {
        super.onListenerConnected()
        if (BuildConfig.DEBUG) Log.d(TAG, "onListenerConnected()")
    }

    override fun onFetchExistingNotifications(notifications: Array<StatusBarNotification>) {
        if (isEnabled) {
            super.onFetchExistingNotifications(notifications)
            mDelegate?.onFetchExistingNotifications(notifications)
        }
    }

    override fun onNotificationPosted(sbn: StatusBarNotification?) {
        if (isEnabled) {
            super.onNotificationPosted(sbn)
            mDelegate?.onNotificationPosted(sbn)
        }
    }

    override fun onNotificationRemoved(sbn: StatusBarNotification?) {
        if (isEnabled) {
            super.onNotificationRemoved(sbn)
            mDelegate?.onNotificationRemoved(sbn)
        }
    }

    private val isEnabled: Boolean
        get() = true
}